### PR TITLE
Minor changes to speed up PointstoPaths algorithm

### PR DESCRIPTION
**Pull request by fuzzysolutions**
## Description
- 1: Replaced time-consuming call on all attributes by a more specific request - responsible for about 90% of running time in my tests.
- 2: Minor speedup by calling items() before the for loop (1s in a large layer).

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 1]
```diff
@@ -144,10 +144,10 @@ def processAlgorithm(self, parameters, context, feedback):
 
             point = f.geometry().constGet().clone()
             if group_field_index >= 0:
-                group = f.attributes()[group_field_index]
+                group = f[group_field_index]    # (1)
```
Comment (luipir) [inline python/plugins/processing/algs/qgis/PointsToPaths.py]: what does it means # (1) or # (2) ?

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 2]
```diff
@@ -165,7 +165,8 @@ def processAlgorithm(self, parameters, context, feedback):
 
         current = 0
         total = 100.0 / len(points) if points else 1
-        for group, vertices in list(points.items()):
+        verlist = list(points.items()) # (2)
```
Comment (luipir) [inline python/plugins/processing/algs/qgis/PointsToPaths.py]: what would be the optimization here? I can't figure it out, sorry!

Comment (nyalldawson) [reply python/plugins/processing/algs/qgis/PointsToPaths.py]: I think the problem here is calling `list(dict.items())` -- that's an anti pattern. Do you see the same results if you change this just to:

    for group, vertices in points.items():

**Comment by nyalldawson** [comment]
> responsible for about 0% of running time in my tests.

Is this a typo?

File: python/plugins/processing/algs/qgis/PointsToPaths.py (hunk 1 above)
Comment (fuzzysolutions) [reply python/plugins/processing/algs/qgis/PointsToPaths.py]: I just labeled the changes mentioned in the pull request description 1 and 2.

Comment (nyalldawson) [reply python/plugins/processing/algs/qgis/PointsToPaths.py]: Right - these will need to be removed, as they have no meaning outside of this pr discussion

**Comment by nyalldawson** [comment]
@fuzzysolutions -- I've pulled your commit into #8381, and indentified a few other places where this same mistake was made. Thank you!

**Pull request closed without merging**